apps/demo/src/app/cart/components/cart/cart.component.spec.ts

```diff
@@ -1,75 +1,157 @@
     TestBed.configureTestingModule({
       declarations: [
-        CartComponent,
         WrapperComponent,
-        MockCartItemComponent
+        MockCartItemsComponent,
+        MockCartSidebarComponent,
+        MockCartItemCountComponent,
+        CartComponent
+      ],
+      providers: [
+        provideMockStore({})
       ]
     })
-    .compileComponents();
+      .compileComponents();
   }));
 
   beforeEach(() => {
     fixture = TestBed.createComponent(WrapperComponent);
     wrapper = fixture.componentInstance;
+    store = TestBed.get(Store);
+    wrapper.cartValue = cart;
+    component = fixture.debugElement.query(By.css('demo-cart')).componentInstance;
+
+    cartItemCountSelector = store.overrideSelector(cartSelector.selectCartItemCount, stubSelectCartItemCount);
+    cartEmptySelector = store.overrideSelector(cartSelector.isCartEmpty, stubIsCartEmpty);
+
+    cartItemsElement = fixture.debugElement.query(By.css('demo-cart-items'));
+    cartItemsComponent = cartItemsElement.componentInstance;
 
-    wrapper.cartValue = mockCart;
+    cartItemCountElement = fixture.debugElement.query(By.css('demo-cart-item-count'));
+    cartItemCountComponent = cartItemCountElement.componentInstance;
+
+    cartSidebarElement = fixture.debugElement.query(By.css('demo-cart-sidebar'));
+    cartSidebarComponent = cartSidebarElement.componentInstance;
 
     fixture.detectChanges();
-    cartItems = fixture.debugElement.queryAll(By.css('demo-cart-item'));
-    cart = fixture.debugElement.query(By.css('demo-cart')).componentInstance;
+  });
+
+  afterAll(() => {
+    store.resetSelectors();
   });
 
   it('should create', () => {
-    expect(cart).toBeTruthy();
+    expect(component).toBeTruthy();
   });
 
-  it('renders a cart', () => {
-    expect(fixture.debugElement.query(By.css('.demo-cart'))).not.toBeNull();
+  it('should be able to take cart as input', () => {
+    expect(component.cart).toEqual(cart);
   });
 
-  it('can be passed a Cart object', () => {
-    expect(cart.cart).toEqual(mockCart);
+  describe('on <demo-cart-items>', () => {
+    it('should pass down the cart', () => {
+      expect(cartItemsComponent.cart).toEqual(cart);
+    });
   });
 
-  it('renders a <demo-cart-item> for every cart.items', () => {
-    expect(cartItems.length).toEqual(mockCart.items.length);
+  describe('on <demo-cart-sidebar>', () => {
+    it('should pass down the cart', () => {
+      expect(cartSidebarComponent.cart).toEqual(cart);
+    });
+
+    it('should set isCartEmpty to false', () => {
+      const val = false;
+      cartEmptySelector.setResult(val);
+      store.setState({});
+      fixture.detectChanges();
+
+      expect(cartSidebarComponent.isCartEmpty).toEqual(val);
+    });
+
+    it('should set isCartEmpty to true', () => {
+      const val = true;
+      cartEmptySelector.setResult(val);
+      store.setState({});
+      fixture.detectChanges();
+
+      expect(cartSidebarComponent.isCartEmpty).toEqual(val);
+    });
```

Comment:
These tests test the same thing, because the component has no concept of what the `isCartEmpty` value is. So only need one of these tests